R/data.table.R

```diff
@@ -1881,17 +1881,58 @@ chmatch2 <- function(x, table, nomatch=NA_integer_) {
 #    x
 #}
 
-
-as.matrix.data.table <- function(x,...)
-{
-  dm <- dim(x)
-  cn <- names(x)
+as.matrix.data.table <- function(x, rownames, ...) {
+  rn <- NULL
+  rnc <- NULL
+  if (!missing(rownames)) { # Convert rownames to a column index if possible
+    if (is.null(rownames)) {
+      warning("rownames is NULL, ignoring rownames")
+    } else if (length(rownames) != 1) {
+      stop("rownames must be a single column in x")
+    } else if (is.na(rownames)) {
+      warning("rownames is NA, ignoring rownames")
+    } else if (is.logical(rownames) && !isTRUE(rownames)) {
+      warning("rownames is FALSE, ignoring rownames")
+    } else if (!(is.logical(rownames) || is.character(rownames) || is.numeric(rownames))) {
+      # E.g. because rownames is some sort of object that cant be converted to a column index
+      stop("rownames must be TRUE, a column index, or a column name in x")
+    } else {
+      if (is.logical(rownames) && isTRUE(rownames)) {
```

Comment:
`isTRUE(rownames)` is sufficient?

Reply:
You're right. I've changed this statement to `identical(rownames, TRUE)` which I think is clearer (and is used elsewhere in data.table.R).